### PR TITLE
Set 'extra_float_digits' to 2 to avoid float8 overflow

### PR DESCRIPTION
**Pull request by seryozhasmirnov**
When analyze for table is performed master segment calls
gp_acquire_sample_rows() helper function on each segment.
That eventually calls float8out function on segment to
converts float8 number to a string with snprintf:

int	ndig = DBL_DIG + extra_float_digits;

snprintf(ascii, MAXDOUBLEWIDTH + 1, "%.*g", ndig, num);

When ndig is 15 the maximum float8 value 1.7976931348623157e+308 is
rounded to "1.79769313486232e+308" that has no representation.
This why extra_float_digits should be set at least to 2.

And on master acquire_sample_rows_dispatcher function
process gp_acquire_sample_rows result and eventually float8in
function is called to convert string to float8 with strtold:
val = strtold(num, &endptr);

This is where overflow for "1.79769313486232e+308" happens but works
fine for "1.7976931348623157e+308".

If table has a filed of double precision type like below:

# create table t (a double precision);

Insert of maximum float8 value 1.7976931348623157e+308 fails as
it is rounded to 1.79769313486232e+308 that has no representation.

# insert into t values (1.7976931348623157e+308);
ERROR:  value out of range: overflow
#

The workaround is to set 'extra_float_digits' GUC value to 2.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Comment by leskin-in** [comment]
As the [tests](https://ci.arenadata.io/job/gpdb/job/PR-211/1/execution/node/59/log/?consoleFull) show, `incremental_analyze` test case must be updated.

**Comment by seryozhasmirnov** [comment]
> As the [tests](https://ci.arenadata.io/job/gpdb/job/PR-211/1/execution/node/59/log/?consoleFull) show, `incremental_analyze` test case must be updated.

Yes, thanks for the note. Updated.

File: src/backend/commands/analyze.c [hunk 1]
```diff
@@ -2571,7 +2571,9 @@ acquire_sample_rows_dispatcher(Relation onerel, bool inh, int elevel,
 	 * may result in different behaviour under different acl configuration.
 	 */
 	initStringInfo(&str);
-	appendStringInfo(&str, "select pg_catalog.gp_acquire_sample_rows(%u, %d, '%s');",
+	appendStringInfo(&str, "select gp_acquire_sample_rows from "
+				"(select set_config('extra_float_digits', '2', true)) sc, "
```
Comment (leskin-in) [inline src/backend/commands/analyze.c:2575]: @darthunix what do you think about this? Is `set_config` guaranteed to affect only the current query?

Comment (seryozhasmirnov) [reply src/backend/commands/analyze.c:2575]: https://www.postgresql.org/docs/9.4/functions-admin.html
> set_config sets the parameter setting_name to new_value. If is_local is true, the new value will only apply to the current transaction. If you want the new value to apply for the current session, use false instead. The function corresponds to the SQL command SET. 

But anyway I am working on getting `gp_acquire_sample_rows` result in binary mode.

**Comment by seryozhasmirnov** [comment]
Closed as https://github.com/arenadata/gpdb/pull/218 provides better solution.

**Pull request closed without merging**